### PR TITLE
Raytasker initial PR

### PR DESCRIPTION
**Pull request by mj0lken**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/raytasker/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "title": "Raytasker",
+  "description": "Raycast google tasks extension",
+  "icon": "raytasker-icon.png",
+  "license": "MIT",
+  "author": "mj0lken",
+  "commands": [
+    {
+      "name": "raytasker",
+      "title": "Raytasker",
+      "description": "Google tasks extension for Raycast",
```
Comment (pernielsentikaer) [inline extensions/raytasker/package.json:13]:
```suggestion
      "title": "Manage Task List",
      "description": "Google Tasks",
```

File: extensions/raytasker/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "raytasker",
+  "title": "Raytasker",
```
Comment (pernielsentikaer) [inline extensions/raytasker/package.json:4]: Then extension should maybe be called Google Tasks instead

File: extensions/raytasker/src/components/dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+export function ListDropdown(props: DDProps) {
+  useEffect(() => {
+    (async () => {
+      props.chooseList(props.chosenList);
+    })();
+  }, [props.chosenList]);
+
+  return (
+    <List.Dropdown
+      tooltip="Select list"
```
Comment (pernielsentikaer) [inline extensions/raytasker/src/components/dropdown.tsx:14]:
```suggestion
      tooltip="Select List"
```

File: extensions/raytasker/src/components/form.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+export function EditTaskForm(props: FormProps) {
+  const { pop } = useNavigation();
+  const [isChanged, setIsChanged] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  return (
+    <Form
+      isLoading={isLoading}
+      navigationTitle={"Task"}
```
Comment (pernielsentikaer) [inline extensions/raytasker/src/components/form.tsx:14]: Please don't change the [navigationTitle](https://developers.raycast.com/basics/prepare-an-extension-for-store#navigation-title) in the root command; it's getting the title from the command.

File: extensions/raytasker/src/components/form.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                    .reduce((a, k) => ({ ...a, [k]: values[k] }), {})) as Task;
+
+                  const res = await google.patchTask(
+                    props?.task?.id ? props.task.id : "",
+                    patchTask.list ? patchTask.list : props.currentList,
+                    patchTask as Task
+                  );
+
+                  if (res.status == 200) {
+                    showToast({ title: `Task updated`, style: Toast.Style.Success });
```
Comment (pernielsentikaer) [inline extensions/raytasker/src/components/form.tsx:66]:
```suggestion
                    showToast({ title: `Task Updated`, style: Toast.Style.Success });
```

File: extensions/raytasker/src/raytasker.tsx [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+  // async function getLists(): Promise<TaskList[]> {
+  //   const lists = await google.fetchLists()
+  //   setAllTaskLists(lists)
+  //   if (!currentList) {
+  //     setChosenList(lists[0].id)
+  //   }
+  //   filterTasks(currentList)
+  //   return lists
+  // }
+
```
Comment (pernielsentikaer) [inline extensions/raytasker/src/raytasker.tsx:52]: Is this just leftovers?

File: extensions/raytasker/src/raytasker.tsx [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+                </List.Item.Detail.Metadata>
+              }
+            />
+          }
+          actions={
+            <ActionPanel>
+              <ActionPanel.Section>
+                <Action
+                  icon={{ source: Icon.Checkmark, tintColor: Color.Blue }}
+                  title="Mark complete"
```
Comment (pernielsentikaer) [inline extensions/raytasker/src/raytasker.tsx:195]:
```suggestion
                  title="Mark Complete"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @mj0lken 👋 

Thanks for your contribution 💯 

I now have the extension running and have some questions and suggestions, could you please look into this?

- When no tasks are on the list there is no way to create a new task since there are no action panel items.

- Maybe hide `Due` if there is no due date set?

- There is an `Icon.Dot` in front of the title in the metadata pane, it looks funny, maybe we should remove it?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by danielsequeira** [comment]
@mj0lken ping here 🙂

**Comment by pernielsentikaer** [comment]
I'm closing this due to inactivity; feel free to open it again or comment in the thread when you're ready to continue working on it 🙂

You can also catch me in [Slack](https://www.raycast.com/community) if you want to discuss 😄

**Pull request closed without merging**